### ISSUE TITLE
Feature Request: Parent/Child "Nested" Tag Relationships

### ISSUE THREAD

**Issue by a2brew** [sysadminsmedia/homebox#112]
### What is the problem you are trying to solve with this feature?

Today, items must be tagged manually, with each applicable tag. This can lead to a somewhat disorganized sprawl of tags, some of which are always 'paired' with each other anyway.

### What is the solution you are proposing?

I think it would be valuable to be able to designate a tag as a 'parent' tag & add 'child' tags to the parent, so if an item is tagged with the child tag, it also shows up in reports/search results for the parent tag as well.

### What alternatives have you considered?

Manual tagging of such 'child/parent' tags individually

### Additional context

Example:

- Parent Tag: Electronics & Accessories
  - Hard Drives
    - Backup Hard Drive
    - Production Hard Drive
    - Spare Hard Drive
  - Computers
    - Laptops
    - Servers

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [ ] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [X] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by Jim-Duke** [comment]
I don't think that it ever makes sense for it to not inherit the parents tag.  Essentially we're talking about creating a taxonomy using tags.  I'll look into doing this as a first effort.

**Comment by zenodotus280** [comment]
Just curious if this has been looked at yet. Nested locations and nested items are great and make good use of the DB relative to a spreadsheet. Nested tags would be icing on the cake!

**Comment by fazalmajid** [comment]
@tonyaellie This is one of the two features I need to migrate to Homebox from my current OmniOutliner file (the other one being support for multiple currencies). I'd be happy to take this on, but rather than building a complex hierarchical UI like for Locations, I wonder if it would make more sense to simply enter tags using an agreed-upon hierarchical delimiter, e.g. / (for instance Electronics & Accessories / Hard Drives / Backup Hard Drive), which would be expanded internally. We can add auto-complete to the UI to make things easier. What do you think?

Update: someone seems to have already started in #1327

**Comment by tonyaellie** [comment]
> @tonyaellie This is one of the two features I need to migrate to Homebox from my current OmniOutliner file (the other one being support for multiple currencies). I'd be happy to take this on, but rather than building a complex hierarchical UI like for Locations, I wonder if it would make more sense to simply enter tags using an agreed-upon hierarchical delimiter, e.g. / (for instance Electronics & Accessories / Hard Drives / Backup Hard Drive), which would be expanded internally. We can add auto-complete to the UI to make things easier. What do you think?
> 
> Update: someone seems to have already started in #1327

Hi @fazalmajid me and Matt are working on it, we are struggling with getting the ux right so we are welcome to new ideas, the current thinking is that you'll be able to set a parent for tags and then when you look at say an item with tags it will show the selected tags and then with some kind of highlight/outline show all the ancestors of the selected tags as "ghost tags", you would then be able to search by any of the tag or ghost tags

**Comment by fazalmajid** [comment]
@tonyaellie FWIW, here is how Cultured Code's Things to-do list app handles it. They are generally considered best-in-class for UX:

https://majid.info/images/reddit/ScreenRecording_02-23-2026%2000-57-42_1.MP4

(obviously Homebox is not a pastry, it's just to show the mechanism of creating a new tag in an existing hierarchy). You will note the filtering box just shows the tag without its parents (what you call ghost tags) as no item is tagged with a parent tag).

**Comment by tonyaellie** [comment]
@fazalmajid 
I don't hate that implement, I'm just not sure how well it would work for say the item card view, it doesn't seem the most space efficient

**Comment by fazalmajid** [comment]
@tonyaellie an alternative is how Lightroom does it. It allows you to type keywords in a text box, use / as delimiters, commas to enter multiple keywords, with autocompletion to make for fast efficient data entry in a professional DAM app.

https://www.youtube.com/watch?v=3cgLw5RsKSQ&embeds_referring_euri=https%3A%2F%2Fwww.perplexity.ai%2F&embeds_referring_origin=https%3A%2F%2Fwww.perplexity.ai&source_ve_path=Mjg2NjY&theme=dark

at the same time, they have a panel to show most recently used tags/keywords for efficiency, and a full hierarchical list for when you need it.

**Comment by Saturo78** [comment]
I’d like to suggest a simplified implementation approach that might reduce complexity significantly.

From my perspective, the nested structure does not necessarily need to be reflected in the UI. Tags could remain visually “flat” exactly as they are today.

Instead, the only functional change would be:
If an item has a child tag, it should also be found when searching/filtering by its parent tag.

In other words, the parent-child relationship would exist purely at the data/query level.

**Comment by dennyblokland** [comment]
I think this feature was included with the recent release of 0.25, please correct me if I'm wrong.

**Comment by tankerkiller125** [comment]
@dennyblokland You are correct it was included with v0.25.0

**Comment by TariqMK** [comment]
I cant see any documentation for this feature, is there a guide? I cant seem to create any nested tags
